Makefile

```diff
@@ -1,25 +1,47 @@
+# Directory of Makefile
+export ROOT_DIR:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+
 GINKGO?="github.com/onsi/ginkgo/v2/ginkgo"
 
 BUILD_DIR?=./build
 
 PKG?=./pkg/... ./internal/...
 COVER_PKG?=github.com/suse/elemental/...
+INTEG_PKG?=./tests/integration/...
 
 GO_MODULE?=$(shell go list -m)
-GO_FILES=$(shell find ./ -name '*.go' -not -name '*_test.go')
+# Exclude files in ./tests folder
+GO_FILES=$(shell find ./ -path ./tests -prune -o -name '*.go' -not -name '*_test.go' -print)
 GO_FILES+=./go.mod
 GO_FILES+=./go.sum
 
 GIT_COMMIT?=$(shell git rev-parse HEAD)
 GIT_COMMIT_SHORT?=$(shell git rev-parse --short HEAD)
 GIT_TAG?=$(shell git describe --candidates=50 --abbrev=0 --tags 2>/dev/null || echo "v0.0.1" )
+VERSION?=$(GIT_TAG)-g$(GIT_COMMIT_SHORT)
 
 LDFLAGS:=-w -s
 LDFLAGS+=-X "$(GO_MODULE)/internal/cli/version.version=$(GIT_TAG)"
 LDFLAGS+=-X "$(GO_MODULE)/internal/cli/version.gitCommit=$(GIT_COMMIT)"
 
 GO_BUILD_ARGS?=-ldflags '$(LDFLAGS)'
 
+# Used to build the OS image
+DOCKER?=docker
+FLAVOR?=green
```

Comment:
nit: I am not sure we need the FLAVOR concept here... Probably this can be dropped or set to `tumbleweed`. I don't think we need a distinction such as green, orange or blue in here.